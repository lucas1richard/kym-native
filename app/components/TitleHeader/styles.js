import { StyleSheet } from 'react-native';
import text from 'styles/Text';
import { h3 } from 'styles/Headings';

const styles = StyleSheet.create({
  wrapper: {
    marginBottom: 30
  },
  title: h3({
    fontWeight: 'bold'
  }),
  text: text({
    textAlign: 'center'
  }, 'halfOpacityBlue')
});

export default styles;
