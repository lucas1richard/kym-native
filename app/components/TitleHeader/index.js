/**
* TitleHeader
*/
import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import styles from './styles';

function TitleHeader({ title, description }) {
  return (
    <View style={styles.wrapper}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.text}>{description}</Text>
    </View>
  );
}

TitleHeader.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string
};

export default TitleHeader;
