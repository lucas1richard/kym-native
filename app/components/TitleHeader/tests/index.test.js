import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TitleHeader from '../';

describe('<TitleHeader />', () => {
  const renderedComponent = (props = {}) => shallow(<TitleHeader {...props} />);

  const props = {
    title: 'Title',
    description: 'Description'
  };
  it('should render without blowing up', () => {
    expect(toJson(renderedComponent(props))).toMatchSnapshot();
  });
});
