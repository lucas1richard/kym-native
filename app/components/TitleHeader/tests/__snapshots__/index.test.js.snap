// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleHeader /> should render without blowing up 1`] = `
<styled.div>
  <Styled(styled.h3)>
    Title
  </Styled(styled.h3)>
  <styled.div
    color="halfOpacityBlue"
  >
    Description
  </styled.div>
</styled.div>
`;
